tests/unit/test_pebble.py

```diff
@@ -0,0 +1,176 @@
+                    "services": {
+                        "mockServiceOne": {"override": "replace", "command": "foo"},
+                        "mockServiceTwo": {"override": "merge", "command": "bar"},
+                    },
+                },
+                (
+                    "summary: mock summary"
+                    "{n}"
+                    "description: mock description"
+                    "{n}"
+                    "services:"
+                    "{n}"
+                    "  mockServiceOne:"
+                    "{n}"
+                    "    override: replace"
+                    "{n}"
+                    "    command: foo"
+                    "{n}"
+                    "  mockServiceTwo:"
+                    "{n}"
+                    "    override: merge"
+                    "{n}"
+                    "    command: bar"
+                    "{n}"
+                ).format(n=os.linesep),
+            ),
+            # Test Case 3:
+            # If there are more files that are not layers, they are ignored.
+            (
+                ["2-bad-layer.yaml", "001-good-layer.yml", "not-a-layer"],
+                "002",
+                {
+                    "summary": "mock summary",
+                    "description": "mock description",
+                    "services": {
+                        "mockServiceOne": {"override": "replace", "command": "foo"}
+                    },
+                },
+                (
+                    "summary: mock summary"
+                    "{n}"
+                    "description: mock description"
+                    "{n}"
+                    "services:"
+                    "{n}"
+                    "  mockServiceOne:"
+                    "{n}"
+                    "    override: replace"
+                    "{n}"
+                    "    command: foo"
+                    "{n}"
+                ).format(n=os.linesep),
+            ),
+        ],
+    )
+    def test_define_pebble_layer(
+        self,
+        tmp_path,
+        existing_layers,
+        expected_new_layer_prefix,
+        layer_content,
+        expected_layer_yaml,
+    ):
+        # pylint: disable=too-many-locals
+        mock_rock_name = "my-rock"
+
+        # Mock the base layer dir just to test the detection of existing
+        # Pebble layers
+        mock_base_layer_dir = tmp_path / "base"
+        tmp_base_layer_dir = mock_base_layer_dir / "var/lib/pebble/default/layers/"
+        tmp_base_layer_dir.mkdir(parents=True)
+        for layer in existing_layers:
+            tmp_layer_file = tmp_base_layer_dir / layer
+            tmp_layer_file.touch()
+
+        mocked_data = {"writes": ""}
+
+        def mock_write(s):
+            mocked_data["writes"] += s
+
+        m = mock_open()
+        pebble_obj = Pebble()
+        with patch("builtins.open", m) as f:
+            with patch("pathlib.Path.mkdir") as local_mock_mkdir:
+                m.return_value.write = mock_write
```

Comment:
I understand that opinions on this diverge, but I think this is too much mocking. Can't we just call the method on a temp path and check the contents afterwards?